### DISCUSSION TITLE
how can i call between Endpoint  and dialog.ts?   and Pass paramter

### DISCUSSION THREAD

**Discussion by ra-bedo11** [serenity-is/Serenity#6334] [Help]
This My Code in
Endpoint.cs

in this end point return Qte for Product

        [HttpPost]
        public int GetQteProduct(int id)
        {
            Serene3_Default_v1Context db = new Serene3_Default_v1Context();
            int valueche = (int)db.TblProducts.SingleOrDefault(w => w.Id == id).ProductQte.Value;
            return valueche;
        }
///////////

how to get Qte in Dialog.ts from EndPoint The Same Product ???
and this my code in Dialog.ts

        private GetQteProduct(pro: Number) {
            what to write here  ????????????????????????????????
        }

**Comment by ra-bedo11** [comment]
؟؟؟؟؟

**Comment by patriciomh1965** [comment]
Hello

Try this code if it can help you

XYZRow.cs
========
....
....
[DisplayName("Numero Carga"), Column("enc_numero"), Identity]
        public System.Int32? EncNumero
        {
            get { return Fields.EncNumero[this]; }
            set { Fields.EncNumero[this] = value; }
        }
....
...




XYZDialog.ts
=========
....
...
  //  deshabilitar botones        
        protected getToolbarButtons(): Serenity.ToolButton[] {
            let buttons = super.getToolbarButtons();
            buttons.splice(Q.indexOf(buttons, x => x.cssClass == "apply-changes-button"), 1);

            let newBtnDos: Serenity.ToolButton =
            {
                title: "Emite a MDC",
                hint: "Permite enviar el archivo a MDC",
                separator: false,
                icon: 'fa fa-upload text-blue',
                cssClass: 'aplicar-button',
                onClick: (e) => {
                    Q.confirm(
                        "Esta Seguro de Continuar con la Operacion?",
                        () => {                            
                            Q.serviceRequest<any>('BaseDatos/ImportarCarga/EmiteMDC',
                                {
                                    MyCarga: this.form.EncNumero.value,
                                }, response => {
                                    Q.notifySuccess(response.MyTestValue, "Operacion Exitosa", { timeOut: 5000 });

                                    Serenity.SubDialogHelper.triggerDataChange(this);   // *** Refreshing the parent grid ***
                                    this.dialogClose();    // *** Closing the dialog ***
                                });
                        },
                        {
                            onNo: () => {
                                Q.notifyInfo("Cancelo Operacion.....");
                                return;
                            },
                            onCancel: () => {
                                Q.notifyError("Se cancela la operación. Lo intentare de nuevo ?");
                                return;
                            },
                            title: "Envio masivo a estado de edición de transaccion"
                        });
                }
            }
            buttons.push(newBtnDos);

            return buttons;
        }
...
...


XYZEndPoint.cs
==========
...
...
 public ServiceResponse EmiteMDC(MYResponse request)
        {
            var Sql = "";
            int cantidad = 0;
            // Actualiza el estado de la carga
            Sql = "UPDATE ta_web_arc_enc SET enc_estado=9 WHERE enc_numero= " + request.MyCarga;
            cantidad = ApoloServices.GrabaCambios.GetGrabaCambios(Sql);

            // contabiliza la cantidad a inyectar a MDC 
            Sql = "SELECT ISNULL(COUNT(*),0) FROM ta_web_arc_det (NOLOCK) WHERE det_encabezado= " + request.MyCarga;
            cantidad = ApoloServices.GrabaCambios.GetGrabaCambios(Sql);

            MYResponse dto = new MYResponse();
            dto.MyTestValue = "Se han enviado " + cantidad + " transacciones ";
            return dto;
        }

        public class MYResponse : ServiceResponse
        {
            public string MyTestValue = "Ok";
            public int MyCarga;
            public string MyRespuesta = "Ok";
        }
